homeassistant/components/private_ble_device/device_tracker.py

```diff
@@ -0,0 +1,77 @@
+"""Tracking for bluetooth low energy devices."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+
+from homeassistant.components import bluetooth
+from homeassistant.components.device_tracker import SourceType
+from homeassistant.components.device_tracker.config_entry import BaseTrackerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_HOME, STATE_NOT_HOME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import BasePrivateDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Load Device Tracker entities for a config entry."""
+    async_add_entities([BasePrivateDeviceTracker(config_entry)])
+
+
+class BasePrivateDeviceTracker(BasePrivateDeviceEntity, BaseTrackerEntity):
+    """A trackable Private Bluetooth Device."""
+
+    _attr_should_poll = False
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, str]:
+        """Return extra state attributes for this device."""
+        attributes = {}
+
+        if self._last_info:
+            attributes["current_address"] = self._last_info.address
+            if self._last_info.source:
+                attributes["source"] = self._last_info.source
+
+        return attributes
```

Comment:
```suggestion
        if self._last_info:
            return {
                "current_address": self._last_info.address,
                "source": self._last_info.source,
            }
        return {}
```


source should always be defined